docs/source-key.md

```diff
@@ -0,0 +1,127 @@
+
+#### Incorrect
+- `feedback_settings` is written by portal-report, but the settings are specific to an assignment in the portal. The sourceKey is `authoring.concord.org`. This is incorrect with this design goal, it should be `learn.concord.org`.
+
+## Possible Improvement: Replace sourceKey with rootKey
+Here is an idea of way to improve the ambiguous use of sourceKeys above.
+
+A single rootKey would be shared by AP, portal-report, LARA, and researcher report.
+The rootKey would default to something like 'production', but could be overridden for staging
+and development testing.
+
+All data would have to change. AP, portal-report, LARA, and researcher report would need to support this new approach, and ideally also be backward compatible with old approach so we can do the migration easily.
+
+### What about the access rules?
+The source is not currently checked by any of the access rules.
+
+### What about conflicting documents?  
+If we don't use the source then the activity-player and LARA answer documents will be mixed in the same folder.
+
+The answer documents currently include a `tool_id`. In the AP this domain plus the path of the url of the running the activity player so it also includes the version or branch in. LARA always adds this to the answers, and it typically should be `https://authoring.concord.org` or a developer might set it to something else for testing purposes.
+
+So this is a way to differentiate between the two types of answer documents.
+
+The question_id in the answer documents is not globably unique by itself, but combining it with the resource_url should uniquely identify the question from the authored content. The same question might be shared by some LARA and AP answers, so a search of answers using just `question_id` and `resource_url` could turn up some answer documents from both systems. However, any reporting system should also be using a `resource_link_id` and/or `run_key` these fields should constrain the answers to a single system.  There are cases that come up with migration where some answers from both systems might share a `resource_link_id`, but that is intentional in the case of a migration.
+
+Perhaps a researcher wants to look at answers to a question across many classes. In that case the query would be:
+`resource_url = https://example.com and question_id = 234`
+That could then pick up answers from different systems. That is hopefully what the researcher would want beause it indicates the same question was given in multiple systems. If a researcher wants to limit the search to just answers created by a particular runtime they'd need to include the `tool_id` in the query.
```

Comment:
```suggestion
That could then pick up answers from different systems. That is hopefully what the researcher would want because it indicates the same question was given in multiple systems. If a researcher wants to limit the search to just answers created by a particular runtime they'd need to include the `tool_id` in the query.
```